services/director-v2/tests/unit/test_modules_director_v0.py

```diff
@@ -2,30 +2,174 @@
+from pathlib import Path
+from typing import Callable, List, Tuple
+
 import pytest
 import respx
-from fastapi import status
+from fastapi import FastAPI, status
+from models_library.services import ServiceDockerData, ServiceKeyVersion
+from simcore_service_director_v2.models.schemas.services import ServiceExtras
+from simcore_service_director_v2.modules.director_v0 import DirectorV0Client
+
+
+@pytest.fixture(autouse=True)
+def minimal_director_config(project_env_devel_environment, monkeypatch):
+    """set a minimal configuration for testing the director connection only"""
+    monkeypatch.setenv("DIRECTOR_ENABLED", "1")
+    monkeypatch.setenv("POSTGRES_ENABLED", "0")
+    monkeypatch.setenv("CELERY_ENABLED", "0")
+    monkeypatch.setenv("REGISTRY_ENABLED", "0")
 
 
 @pytest.fixture
-def mocked_director_v0_service_api(minimal_app):
+def mocked_director_v0_service_api(minimal_app, entrypoint, exp_data, resp_alias):
     with respx.mock(
         base_url=minimal_app.state.settings.director_v0.base_url(include_tag=False),
         assert_all_called=False,
         assert_all_mocked=True,
     ) as respx_mock:
         # lists services
         respx_mock.get(
+            urllib.parse.unquote(entrypoint),
+            content=exp_data,
+            alias=resp_alias,
+        )
+
+        yield respx_mock
+
+
+def _get_list_services_calls() -> List[Tuple]:
+    return [
+        (
+            "/v0/services",
+            status.HTTP_200_OK,
+            {"data": ["service1", "service2"]},
+            "list_all_services",
+        ),
+        (
             "/v0/services?service_type=computational",
-            content={"data": ["service1", "service2"]},
-            alias="list_services",
+            status.HTTP_200_OK,
+            {"data": ["service1", "service2"]},
+            "list_computational_services",
+        ),
+        (
+            "/v0/services?service_type=dynamic",
+            status.HTTP_200_OK,
+            {"data": ["service1", "service2"]},
+            "list_dynamic_services",
+        ),
+    ]
+
+
+def _get_service_version_calls() -> List[Tuple]:
+    # TODO: here we see the return value is currently not validated
+    return [
+        (
+            "/v0/services/simcore%2Fservices%2Fdynamic%2Fmyservice/1.3.4",
+            status.HTTP_200_OK,
+            {"data": ["stuff about my service"]},
+            "get_service_version",
         )
+    ]
+
+
+def _get_service_version_extras_calls() -> List[Tuple]:
+    # TODO: here we see the return value is currently not validated
+    return [
+        (
+            "/v0/services/simcore%2Fservices%2Fdynamic%2Fmyservice/1.3.4/extras",
+            status.HTTP_200_OK,
+            {"data": "extra stuff about my service"},
+            "get_service_extras",
+        )
+    ]
+
+
+@pytest.mark.parametrize(
+    "entrypoint,exp_status,exp_data,resp_alias",
+    _get_list_services_calls()
+    + _get_service_version_calls()
+    + _get_service_version_extras_calls(),
```

Comment:
I know it is more verbose but something more readable like name tuples (for the content of the functions) help a lot with going through these. If possible.

Reply:
example?

Reply:
Something similar to what you see below.

```python
from collections import namedtuple

ForwardToDirectorParams = namedtuple(
    "ForwardToDirectorParams", "entrypoint exp_status exp_data resp_alias"
)


def _get_list_services_calls() -> List[Tuple]:
    return [
        ForwardToDirectorParams(
            entrypoint="/v0/services",
            exp_status=status.HTTP_200_OK,
            exp_data={"data": ["service1", "service2"]},
            resp_alias="list_all_services",
        ),
        ForwardToDirectorParams(
            entrypoint="/v0/services?service_type=computational",
            exp_status=status.HTTP_200_OK,
            exp_data={"data": ["service1", "service2"]},
            resp_alias="list_computational_services",
        ),
        ForwardToDirectorParams(
            entrypoint="/v0/services?service_type=dynamic",
            exp_status=status.HTTP_200_OK,
            exp_data={"data": ["service1", "service2"]},
            resp_alias="list_dynamic_services",
        ),
    ]
```